lib/ride.rb

```diff
@@ -1,3 +1,27 @@
 class Ride
+    attr_reader :name,
+                :distance,
+                :loop,
+                :terrain
+
+    def initialize(features)
+        @name = features[:name]
+        @distance = features[:distance]
+        @loop = features[:loop]
+        @terrain = features[:terrain]
+    end
+
+    def loop?
+        @loop
+    end
+
+    def total_distance
+        if @loop
+            actual_distance = distance
+        else
+            actual_distance = distance * 2
+        end
+        actual_distance
```

Comment:
Nice work! You could also just use the values `distance` and `distance * 2` without creating the extra variable